flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/NetworkBufferPool.java

```diff
@@ -181,6 +196,19 @@ public void recycle(MemorySegment segment) {
 				if (segment != null) {
 					segments.add(segment);
 				}
+
+				if ((System.nanoTime() - startTimeInNanos) / 1_000_000 >= requestSegmentsTimeoutInMillis &&
+						segments.size() < numberOfSegmentsToRequest) {
+					throw new IOException(String.format("Insufficient number of network buffers: " +
```

Comment:
And before the real request action, it has checked that `numTotalRequiredBuffers + numberOfSegmentsToRequest > totalNumberOfMemorySegments` above, can this situation happen?

Reply:
This might happen if the configured number of buffers on TM is less than the `required` one.  On TM, the number of buffers in local buffer pool is defined by two values, namely `[required, max)`, and each RemoteInputChannel will also counts its exclusive buffers as `required` buffers. If the sum of the `required` number of buffers has exceeded the number of buffers in the NetworkBufferPool, the above situation happens.